lib/aliquot/validator.rb

```diff
@@ -47,23 +48,21 @@ def self.to_bool(lbd)
       # the above Base64? predicate and use the following simpler one:
       #predicate(:strict_base64?) { |x| !!Base64.strict_decode64(x) rescue false }
 
-      predicate(:pan?) { |x| str?(x) && match_b.call(x, /\A[1-9][0-9]{11,18}\z/) }
+      predicate(:pan?) { |x| match_b.call(x, /\A[1-9][0-9]{11,18}\z/) }
```

Comment:
In other places we accept PANs starting with 0, which I have pointed out in my review PR.

Reply:
Although it isn't stated in https://developer.apple.com/library/archive/documentation/PassKit/Reference/PaymentTokenJSON/PaymentTokenJSON.html#//apple_ref/doc/uid/TP40014929-CH8-SW4 (`applicationPrimaryAccountNumber`) we also validate PANs that way in pedicel: https://github.com/clearhaus/pedicel/blob/a8199390c487a5c8eb0a18e30e2c0db2036613d6/lib/pedicel/validator.rb#L49

Reply:
If you use this lib and you find that you decline transactions because of this, then you could just monkey-patch the heck out of it and make an issue/PR on this repo. If we leave it to the client to remove leading zeros, then, if you (in error) receive a PAN with leading zero then "b0rk".

I'd like to keep it stricter and loosen if necessary (because we'd never tighten it since there's no "process" for noticing it could be stricter) 🙂

The same could be said about `strict_base64`, but our experience from 3DS is that you'd most likely receive non-strict Base64 ... But I'm all for tightening that one too.